### ISSUE TITLE
Feature: Setting to eagerly initialize all beans

### ISSUE THREAD

**Issue by knyttl** [micronaut-projects/micronaut-core#1627]
### Motivation

We are using Kubernetes where application report itself as healthy when everything is started correctly. If new version of application reports OK on /health, it should mean it was correctly initialised and can start serving requests. 

### What happens now

When new app version starts, not all beans/controllers are yet initialised. However, the app reports OK on /health, so Kubernetes replaces the old version with new one.

Because the beans are initialised lazily, if any of the beans are misconfigured, they will fail to initialise once accessed. This is not desired, because the old working application has already been replaced with new one, which is malfunctioning.

Basically, what we need is the /health endpoint to response OK just AFTER all beans are initialised. If any of the beans fail to initialise, the /health endpoint should not respond with OK. Also /healht endpoint should not report healthy during the initialisation process.

### Current state

We can achieve this with manually annotating all beans with `@Context`. However this leads to easily forgetting, especially if there is a lot of people collaborating.

### Desired state

We would like a configuration option, which would eagerly initialise all beans, in a same way like Spring does.

**Comment by jeffbrown** [comment]
Is the request to only do this for `Singleton` beans?  I think the potential for misconfigured beans is still going to exist and potentially won't be resolvable until those beans are actually needed in some cases.

**Comment by knyttl** [comment]
I guess singleton beans would be enough. You are right - in some cases - but at least the inejction of dependencies and init block will eliminate a lot of issues.

**Comment by knyttl** [comment]
For instance now the application can become healthy even without acquiring database connection. The EntityManager cannot initialise, but the app becomes healthy.

**Comment by graemerocher** [comment]
if you simply ran your tests that would not be the case. I assume you are developing a project that contains tests? The `EntityManager` in fact is a `@Context` scoped bean so will be initialized on startup. In general we create `@Context` scope beans for beans that are likely to fail such as connections, and so on. A simple misconfiguration of DI which would be picked up easily in a test is not such as case.

Having said that we will consider it

**Comment by knyttl** [comment]
Another reason is to prepare the application to be able to serve coming requests as quickly as possible. If not all beans are fully initialised, first requests coming to the application will take longer time.

**Comment by alexwilbur** [comment]
We would also love this. We're using AWS lambda and with the new provisioned concurrency feature we'd like (if possible) a simple configuration setting that instructs Micronaut to eagerly initialize all beans so that when AWS puts our (pre-warmed) lambda into service, we no longer observe any initialization overhead

See: 
https://aws.amazon.com/about-aws/whats-new/2019/12/aws-lambda-announces-provisioned-concurrency/
